src/Scyther/Theory/Lexer.x

```diff
@@ -228,18 +225,41 @@ instance Show Keyword where
 -- -----------------------------------------------------------------------------
 -- The input type
 
-
 type AlexInput = (AlexPosn,     -- current position,
                   Char,         -- previous char
+                  [Word8],      -- remaining bytes of current char
                   String)       -- current input string
 
 alexInputPrevChar :: AlexInput -> Char
-alexInputPrevChar (p,c,s) = c
-
-alexGetChar :: AlexInput -> Maybe (Char,AlexInput)
-alexGetChar (p,c,[]) = Nothing
-alexGetChar (p,_,(c:s))  = let p' = alexMove p c in p' `seq`
-                                Just (c, (p', c, s))
+alexInputPrevChar (p,c,bs,s) = c
+
+alexGetByte :: AlexInput -> Maybe (Word8,AlexInput)
+alexGetByte (p,c,(b:bs),s) = Just (b,(p,c,bs,s))
+alexGetByte (p,c,[],[]) = Nothing
+alexGetByte (p,_,[],(c:s))  = let p' = alexMove p c
+                                  (b:bs) = utf8Encode c
+                              in p' `seq`  Just (b, (p', c, bs, s))
+
+-- | Encode a Haskell String to a list of Word8 values, in UTF8 format.
```

Comment:
You might just use <https://hackage.haskell.org/package/utf8-string-1/docs/src/Codec-Binary-UTF8-String.html#encodeChar> instead of implementing it locally. This way, you leave the bit-fiddling to a library.

In general, I'm not sure whether this is the right way to add UTF-8 support to Alex. However, if it works, then it's OK.